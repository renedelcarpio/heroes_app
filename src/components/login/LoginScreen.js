import './LoginScreen.scss';

import { AuthContext } from '../../auth/authContext';
import { types } from '../../types/types';
import { useContext } from 'react';
import { useNavigate } from 'react-router-dom';

const LoginScreen = () => {
	const navigate = useNavigate();
	const { dispatch } = useContext(AuthContext);

	const handleLogin = () => {
		const action = {
			type: types.login,
			payload: { name: 'Santi' },
		};

		dispatch(action);

		const lastPath = localStorage.getItem('lastPath') || '/';

		navigate(lastPath, { replace: true });
	};

	return (
		<div className='login'>
			<h1>Login</h1>
			<hr />
			<button className='button-login' onClick={handleLogin}>
				Login
			</button>
		</div>
	);
};

export default LoginScreen;
