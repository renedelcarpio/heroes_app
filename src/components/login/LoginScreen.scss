@import '../../index.scss';

.login {
  padding: 1rem;
}

.button-login {
  background-color: $color-primary-dc;
  border: 2px solid $color-primary-dc;
  border-radius: 0.5rem;
  color: $color-light;
  font-size: 1.5rem;
  margin: 1.5rem;
  padding: 0.2rem 0.5rem;
  transition: all 0.3s;

  &:hover {
    font-size: 1.8rem;
  }
}
