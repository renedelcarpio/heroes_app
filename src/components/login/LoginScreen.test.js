import { AuthContext } from '../../auth/authContext';
import LoginScreen from './LoginScreen';
import { mount } from 'enzyme';
import { types } from '../../types/types';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useNavigate: () => mockNavigate,
}));

describe('Tests in LoginScreen component', () => {
	const contextValue = {
		dispatch: jest.fn(),
		user: {
			logged: false,
		},
	};

	const wrapper = mount(
		<AuthContext.Provider value={contextValue}>
			<LoginScreen />
		</AuthContext.Provider>
	);

	test('should match with snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should call dispatch and navigate', () => {
		const handleClick = wrapper.find('button').prop('onClick');
		handleClick();

		expect(contextValue.dispatch).toHaveBeenCalledWith({
			type: types.login,
			payload: { name: 'Santi' },
		});
		expect(mockNavigate).toHaveBeenCalledWith('/', { replace: true });

		localStorage.setItem('lastPath', '/dc');
		handleClick();
		expect(mockNavigate).toHaveBeenCalledWith('/dc', { replace: true });
	});
});
