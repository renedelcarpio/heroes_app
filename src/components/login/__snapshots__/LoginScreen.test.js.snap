// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in LoginScreen component should match with snapshot 1`] = `
<div
  className="login"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="button-login"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
